src/chat/entity_button.rs

```diff
@@ -100,71 +101,105 @@ live_design!(
 );
 
 #[derive(Live, Widget, LiveHook)]
-pub struct AgentButton {
+pub struct EntityButton {
     #[deref]
     view: View,
 
     #[rust]
-    agent: Option<MofaAgent>,
-
-    #[live(false)]
-    create_new_chat: bool,
-
-    #[live(false)]
-    select_agent_on_prompt: bool,
```

Comment:
This boolean flags change the widget behavior to do stuff that can be easily done outside. Those behaviors are implemented internally by dispatching global actions which harms re-usability unnecessarily. 

So they were replaced with a simple `.clicked()` method (like normal button's method), and that stuff is handled outside.